src/publish/publishToRegistry.ts

```diff
@@ -53,29 +61,19 @@ export async function publishToRegistry(originalBumpInfo: BumpInfo, options: Bea
   if (prepublishHook) {
     for (const pkg of packagesToPublish) {
       const packageInfo = bumpInfo.packageInfos[pkg];
-      const maybeAwait = prepublishHook(
-        path.dirname(packageInfo.packageJsonPath),
-        packageInfo.name,
-        packageInfo.version
-      );
-      if (maybeAwait instanceof Promise) {
-        await maybeAwait;
-      }
+      await prepublishHook(path.dirname(packageInfo.packageJsonPath), packageInfo.name, packageInfo.version);
     }
   }
 
   // finally pass through doing the actual npm publish command
-  let hasAuthError = false;
-  for (const pkg of packagesToPublish) {
-    if (hasAuthError) {
```

Comment:
This separate hasAuthError check appears to be unnecessary since the end of this `for` loop throws if individual package publishing was unsuccessful for any reason